lib/client/loki.go

```diff
@@ -0,0 +1,155 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/smartcontractkit/chainlink-testing-framework/lib/logging"
+)
+
+// LokiResponse represents the structure of the response from Loki
+type LokiResponse struct {
+	Data struct {
+		Result []struct {
+			Stream map[string]string `json:"stream"`
+			Values [][]interface{}   `json:"values"`
+		} `json:"result"`
+	} `json:"data"`
+}
+
+// LokiLogEntry represents a single log entry with a timestamp and raw log message
+type LokiLogEntry struct {
+	Timestamp string
+	Log       string
+}
+
+// LokiClient represents a client to interact with Loki for querying logs
+type LokiClient struct {
+	BaseURL     string
+	TenantID    string
+	BasicAuth   string
+	QueryParams LokiQueryParams
+	Logger      logging.Logger
+	RestyClient *resty.Client
+}
+
+// LokiQueryParams holds the parameters required for querying Loki
+type LokiQueryParams struct {
+	Query     string
+	StartTime time.Time
+	EndTime   time.Time
+	Limit     int
+}
+
+// NewLokiClient creates a new Loki client with the given parameters, initializes a logger, and configures Resty with debug mode
+func NewLokiClient(baseURL, tenantID, basicAuth string, queryParams LokiQueryParams) *LokiClient {
+	logging.Init()
+
+	logger := logging.GetLogger(nil, "LOKI_CLIENT_LOG_LEVEL")
+	logger.Info().
+		Str("BaseURL", baseURL).
+		Str("TenantID", tenantID).
+		Msg("Initializing Loki Client")
+
+	// Set debug mode for Resty if RESTY_DEBUG is enabled
+	isDebug := os.Getenv("RESTY_DEBUG") == "true"
+
+	restyClient := resty.New().
+		SetDebug(isDebug)
+
+	return &LokiClient{
+		BaseURL:     baseURL,
+		TenantID:    tenantID,
+		BasicAuth:   basicAuth,
+		QueryParams: queryParams,
+		Logger:      logger,
+		RestyClient: restyClient,
+	}
+}
+
+// QueryLogs queries Loki logs based on the query parameters and returns the raw log entries
+func (lc *LokiClient) QueryLogs(ctx context.Context) ([]LokiLogEntry, error) {
+	// Log request details
+	lc.Logger.Info().
+		Str("Query", lc.QueryParams.Query).
+		Str("StartTime", lc.QueryParams.StartTime.Format(time.RFC3339Nano)).
+		Str("EndTime", lc.QueryParams.EndTime.Format(time.RFC3339Nano)).
+		Int("Limit", lc.QueryParams.Limit).
+		Msg("Making request to Loki API")
+
+	// Start tracking request duration
+	start := time.Now()
+
+	// Build query parameters
+	params := map[string]string{
+		"query": lc.QueryParams.Query,
+		"start": lc.QueryParams.StartTime.Format(time.RFC3339Nano),
+		"end":   lc.QueryParams.EndTime.Format(time.RFC3339Nano),
+		"limit": fmt.Sprintf("%d", lc.QueryParams.Limit),
+	}
+
+	// Send request using the pre-configured Resty client
+	resp, err := lc.RestyClient.R().
+		SetContext(ctx).
+		SetHeader("X-Scope-OrgID", lc.TenantID).
+		SetBasicAuth(lc.BasicAuth, "").
```

Comment:
Not sure we can send `login:password` as one string here. Either let's remove it for now or split the field and insert `Login` and `Password` properly.

Reply:
This has been split.